File: src/main/java/scrumweb/dto/fieldcontent/FieldsContentCollector.java
package scrumweb.dto.fieldcontent;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;
import scrumweb.common.FieldsExtractor;
import scrumweb.dto.projectfield.ProjectFieldDto;

import java.util.HashSet;
import java.util.Set;

@NoArgsConstructor
@AllArgsConstructor
@Getter @Setter
@Component
public class FieldsContentCollector implements FieldsExtractor<FieldContentDto> {

    private Set<CheckBoxContainerContentDto> checkBoxContainerContentDtos;
    private Set<InputFieldContentDto> inputFieldContentDtos;
    private Set<ListElementsContainerContentDto> listElementsContainerContentDtos;
    private Set<RadioButtonContainerContentDto> radioButtonContainerContentDtos;
    private Set<TextAreaContentDto> textAreaContentDtos;

    @Override
    public Set<FieldContentDto> extractFields() {
        Set<FieldContentDto> projectFieldDtos = new HashSet<>();
        projectFieldDtos.addAll(checkBoxContainerContentDtos);
        projectFieldDtos.addAll(inputFieldContentDtos);
        projectFieldDtos.addAll(listElementsContainerContentDtos);
        projectFieldDtos.addAll(radioButtonContainerContentDtos);
        projectFieldDtos.addAll(textAreaContentDtos);
        return projectFieldDtos;
    }
}

File: src/frontend/src/app/model/fields-content/FieldsContentCollector.ts
import { CheckBoxContainerContentDto } from './CheckBoxContainerContentDto';
import { InputFieldContentDto } from './InputFieldContentDto';
import { ListElementsContainerContentDto } from './ListElementsContainerContentDto';
import { RadioButtonContainerContentDto } from './RadioButtonContainerContentDto';
import { TextAreaContentDto } from './TextAreaContentDto';

export class FieldsContentCollector {
    checkBoxContainerContentDtos: Array<CheckBoxContainerContentDto> = [];
    inputFieldContentDtos: Array<InputFieldContentDto> = [];
    listElementsContainerContentDtos: Array<ListElementsContainerContentDto> = [];
    radioButtonContainerContentDtos: Array<RadioButtonContainerContentDto> = [];
    textAreaContentDtos: Array<TextAreaContentDto> = [];
}

